confidant/authnz/userauth.py

```diff
@@ -113,13 +114,24 @@ def set_expiration(self):
                 max_expiration = now + datetime.timedelta(seconds=max_lifetime)
                 session['max_expiration'] = max_expiration
 
-    def set_current_user(self, email, first_name=None, last_name=None):
+    def set_current_user(self, email, first_name=None, last_name=None, username=None):
+        if email is None or email == '':
+            raise errors.UserUnknownError('No email provided')
+
         session['user'] = {
             'email': email,
             'first_name': first_name,
             'last_name': last_name,
+            'groups': []
         }
 
+        if username is None:
+            username = email.strip().split('@')[0]
+
+        if app.config.get('USE_GROUPS'):
+            all_groups = grp.getgrall()
+            session['user']['groups'] = [g.gr_name for g in all_groups if email in g.gr_mem]
```

Comment:
I think it would be good if fetching groups was something modular, as it's possible that some folks will get the user's group from a SAML assertion, some will get it from a header, and others may want to use unix groups like this.